edi_oca/models/edi_exchange_record.py

```diff
@@ -237,7 +239,7 @@ def name_get(self):
         for rec in self:
             rec_name = rec.identifier
             if rec.res_id and rec.model:
-                rec_name = rec.record.display_name
+                rec_name = rec.sudo().record.display_name
```

Comment:
Given the change introduced in how the `record` property is defined, I think `sudo()` should be postponed to `record`:
```suggestion
                rec_name = rec.record.sudo().display_name
```